### PR TITLE
gen2 cfu

### PR DESCRIPTION
**Pull request by alanvgreen**
Implementation of CFU for simple cases. Try menu options 3 (project menu) then 5 (test layer 05).

### CODE REVIEW

**Review by tcal-x — APPROVED** [review]

File: proj/hps_accel/Makefile [hunk 1]
```diff
@@ -74,7 +74,7 @@ EXTRA_LITEX_ARGS += --cpu-variant slim+cfu
 
 # gen2 needs special build args and only works on hps
 ifeq '$(GATEWARE_GEN)' '2'
-EXTRA_LITEX_ARGS += --separate-arena --cfu-mport --cpu-variant=slimopt+cfu
+EXTRA_LITEX_ARGS += --separate-arena --cfu-mport
```
Comment (danc86) [inline proj/hps_accel/Makefile:77]: This looks unrelated to this commit, and maybe you didn't intend to include it?

File: soc/hps_proto2_platform.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def __init__(self, platform, sys_clk_freq):
 class Platform(LatticePlatform):
     # The NX-17 has a 450 MHz oscillator. Our system clock should be a divisor
     # of that.
-    clk_divisor = 7
+    clk_divisor = 9
```
Comment (danc86) [inline soc/hps_proto2_platform.py:116]: The commit message for this says "DO NOT MERGE" :-) but maybe you do want to merge this anyway? The fast slew rate should fix SPI flash corruption I guess, but maybe we still need 50MHz system clock to get this to route?

Comment (alanvgreen) [reply soc/hps_proto2_platform.py:116]: I'll keep this for now, until the nextpnr slew rate fixes make it into CFU Playground.

File: proj/hps_accel/Makefile (hunk 1 above)
Comment (alanvgreen) [reply proj/hps_accel/Makefile:77]: Thanks. Removed.

**Review by danc86 — APPROVED** [review]
